pkg/scheduler/objects/application.go

```diff
@@ -1712,6 +1715,17 @@ func (sa *Application) ReplaceAllocation(uuid string) *Allocation {
 	if sa.placeholderData != nil {
 		sa.placeholderData[ph.GetTaskGroup()].Replaced++
 	}
+
+	// check if we have active placeholder allocation left
+	hasReplaceablePlaceholder := false
+	for _, phData := range sa.placeholderData {
```

Comment:
Yes, it is good to set/unset in existing places (appropriate as described). do we need to unset even when app transition to work as non-gang because of "soft" style"?

Reply:
You're right, we have `hasPlaceHolderAllocations := len(sa.getPlaceholderAllocations()) > 0` which can be used for this purpose.